webui/react/src/hooks/useSettings.ts

```diff
@@ -246,53 +251,68 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
         }
       }
     },
-    [user?.id, config.storagePath, settings],
+    [config.storagePath, loadableUser],
   );
 
   const resetSettings = useCallback(
-    async (settingsArray?: string[]) => {
-      if (!settings) return;
+    (settingsArray?: string[]) => {
+      const user = Loadable.match(loadableUser.get(), {
+        Loaded: (cUser) => cUser,
+        NotLoaded: () => undefined,
+      });
+      if (!user) return;
 
       const array = settingsArray ?? Object.keys(config.settings);
-      const newSettings = { ...settings };
 
-      array.forEach((setting) => {
-        let defaultSetting: SettingsConfigProp<T[Extract<keyof T, string>]> | undefined = undefined;
+      stateOb.update((s) => {
+        const news = s.get(config.storagePath);
+        array.forEach((setting) => {
+          let defaultSetting: SettingsConfigProp<T[Extract<keyof T, string>]> | undefined =
+            undefined;
 
-        for (const key in config.settings) {
-          const conf = config.settings[key];
+          for (const key in config.settings) {
+            const conf = config.settings[key];
 
-          if (conf.storageKey === setting) {
-            defaultSetting = conf;
-            break;
+            if (conf.storageKey === setting) {
+              defaultSetting = conf;
+              break;
+            }
           }
-        }
 
-        if (!defaultSetting) return;
+          if (!defaultSetting || !news) return;
 
-        newSettings[setting as keyof T] = defaultSetting.defaultValue;
+          news[setting] = defaultSetting.defaultValue;
+        });
+        return s.set(config.storagePath, news ?? {});
       });
-
-      update(config.storagePath, newSettings);
-
-      await updateDB(newSettings);
-
-      navigate('', { replace: true });
     },
-    [config, update, updateDB, navigate, settings],
+    [config, stateOb, loadableUser],
   );
 
   const updateSettings = useCallback(
-    async (updates: Settings, shouldPush = false) => {
-      if (!settings) return;
-
-      const newSettings = { ...settings, ...updates };
-
-      if (isEqual(newSettings, settings)) return;
+    (updates: Settings) => {
+      stateOb.update((s) =>
+        s.set(
+          config.storagePath,
+          s.get(config.storagePath) === updates
+            ? s.get(config.storagePath) ?? {}
+            : { ...s.get(config.storagePath), ...updates },
+        ),
```

Comment:
`s.get(config.storagePath)` and `updates` are both `Settings` object type, so we cant use  `===` here.
also, the code isnt readable due to one liner. i think we should make it readable